pallets/streaming/src/lib.rs

```diff
@@ -412,26 +416,29 @@ impl<T: Config> Pallet<T> {
         Ok(duration)
     }
 
-    pub fn update_finished_stream_library(account: &AccountOf<T>) -> DispatchResult {
+    pub fn update_finished_stream_library(
+        sender: &AccountOf<T>,
+        recipient: &AccountOf<T>,
+    ) -> DispatchResult {
         let checked_pop =
             |registry: &mut Option<BoundedVec<StreamId, T::MaxStreamsCount>>| -> DispatchResult {
                 let mut r = registry.take().unwrap_or_default();
-                r.as_mut().sort_unstable_by(|a, b| b.cmp(a));
-
                 let len = r.len() as u32;
                 match len {
-                    _x if len >= T::MaxFinishedStreamsCount::get() => {
+                    _x if len > T::MaxFinishedStreamsCount::get() => {
                         if let Some(stream_id) = r.pop() {
                             if Streams::<T>::get(stream_id).is_some() {
```

Comment:
deep nested scope can be improved I guess

Reply:
```rust
if r.len() as u32 > T::MaxFinishedStreamsCount::get() {
    // This is safe because we have ensured that the length is greater than the limit(at least 0)
    let stream_id = r.pop().unwrap();
    if Streams::<T>::get(stream_id).is_none() {
        return Ok(())
    };
    // This is safe because we have ensured that the stream is in storage
    let stream = Streams::<T>::get(stream_id).unwrap();

    // Remove all related storage
    Self::try_remove_stream_library(&stream.sender, stream_id, None)?;
    Self::try_remove_stream_library(
        &stream.recipient,
        stream_id,
        None,
    )?;
    Streams::<T>::remove(stream_id);
}
*registry = Some(r);
```

Reply:
how about this solution? I've tried it before, but there is a cascading relationship between these logics.

Reply:
is it possible to quick return?

Reply:
There will be a lot of quick return.

Reply:
```rust
if r.len() as u32 <= T::MaxFinishedStreamsCount::get() {
    *registry = Some(r);
    return Ok(());
}

// This is safe because we have ensured that the length is greater than the limit(at least 0)
if let Some(stream_id) = r.pop() {
    let stream = Streams::<T>::get(stream_id).ok_or(Error::<T>::InvalidStreamId)?;
    // Remove all related storage
    Self::try_remove_stream_library(&stream.sender, stream_id, None)?;
    Self::try_remove_stream_library(&stream.recipient, stream_id, None)?;
    Streams::<T>::remove(stream_id);
```